contributors/design-proposals/cluster-info-configmap.md

```diff
@@ -0,0 +1,25 @@
+# Persist cluster-wide data in ConfigMap
+
+## Abstract
+
+We want to allow cluster-wide data to be persisted for general purposes, currently for us (service catalog) is to focus on the ability to identify different clusters that access our service broker with a 'cluster-id'.
+
+## Motivation
+
+Currently there is no way for service catalog api server to identify the different clusters that access our service brokers. The goal of this design is to have a unify storage for cluster related information. The information should be retained and survive between component failures. 
+
+## Detailed design
+
+A configmap `cluster-info` is created in the `default` namespace at the starup of the controller. The initial content of the map will be as the following:
+
+```
+apiVersion: v1
+data:
+  cluster-id: uuid-of-the-cluster
+kind: ConfigMap
+metadata:
+  name: cluster-info
+  namespace: default
```

Comment:
the question of use gets into the second aspect, which is the namespace, name, and visibility of this configmap:
* should the cluster id be visible to anyone who wants to see it?
  * if so, the `kube-public` namespace is probably a better choice.
  * if not, the `kube-system` namespace *may* be a better choice (though that depends on who should be authorized to access it)
* what other information do we anticipate being distributed in this configmap? do we anticipate a need for mixed levels of confidentiality/visibility?

as soon as data is available here, the configmap namespace, name, and keys become a de facto API and the magic names we choose are locked. we should think through what will happen when we need to 1) add additional information, 2) have information of varying levels of confidentiality, and 3) evolve the cluster info with mixed versions of clients using the info.

Reply:
Any client that can access the API, inside or outside the cluster, should be able to know the cluster name and/or id. We probably will even want to add it to the downward API.